### PR TITLE
Use `information_schema` as a basis for `describe-database` in Databricks driver

### LINKED ISSUE #48584: Use `information_schema` instead of jdbc methods in `describe-database` in Databricks driver
Parent `:sql-jdbc` implementation of `describe-database` (1) fetches all the schemas from all catalogs and (2) in some circumstances can produce an exception. See [this slack thread](https://metaboat.slack.com/archives/C07L35T7UFQ/p1728318329287749) for the details.

To overcome that, `information_schema` should be used to get the necessary information during sync.

### PR DESCRIPTION
**Pull request by lbrdnk**
Closes #48584

### Description

This PR makes `describe-database` use `information_schema` instead of jdbc `.getSchemas`.

### How to verify

`describe-database` test was extended.

### CODE REVIEW

File: modules/drivers/databricks/test/metabase/test/data/databricks.clj [hunk 1]
```diff
@@ -126,7 +126,7 @@
   [driver dbdef]
   (if *allow-database-deletion*
     ((get-method tx/destroy-db! :sql-jdbc/test-extensions) driver dbdef)
-    (log/warn "`*allow-database-creation*` is `false`. Database removal is suppressed.")))
+    (log/warn "`*allow-database-deletion*` is `false`. Database removal is suppressed.")))
```
Comment (lbrdnk) [inline modules/drivers/databricks/test/metabase/test/data/databricks.clj:129]: Unrelated, typo.

File: src/metabase/driver.clj [hunk 2]
```diff
@@ -735,7 +735,8 @@
                               :schemas                                true
                               :test/jvm-timezone-setting              true
                               :fingerprint                            true
-                              :upload-with-auto-pk                    true}]
+                              :upload-with-auto-pk                    true
+                              :test/dynamic-dataset-loading           true}]
```
Comment (lbrdnk) [inline src/metabase/driver.clj:739]: Thanks @appleby! Setting the default to true for `dynamic-dataset-loading` was missing.

File: modules/drivers/databricks/src/metabase/driver/databricks.clj [hunk 3]
```diff
@@ -44,6 +45,39 @@
...
+
+(defn- describe-database-tables
+  [database]
+  (let [[inclusion-patterns
+         exclusion-patterns] (driver.s/db-details->schema-filter-patterns database)
+        syncable? (fn [schema]
+                    (driver.s/include-schema? inclusion-patterns exclusion-patterns schema))]
+    (eduction
+     (filter (comp syncable? :schema))
+     (sql-jdbc.execute/reducible-query database (get-tables-sql (-> database :details :catalog))))))
```
Comment (metamben) [inline modules/drivers/databricks/src/metabase/driver/databricks.clj:70]: Why doesn't this return a set?

File: modules/drivers/databricks/test/metabase/driver/databricks_test.clj [hunk 4]
```diff
@@ -21,7 +22,27 @@
...
+             (driver/describe-database :databricks (mt/db)))))
+    (testing "`driver/describe-database` returns expected results for `all` schema filters."
+      (let [actual-tables (driver/describe-database :databricks (-> (mt/db)
+                                                                    (update :details dissoc :schema-filters-patterns)
+                                                                    (update :details assoc  :schema-filters-type "all")))]
+        (testing "tables from multiple schemas were found"
+          (is (contains? (:tables actual-tables) {:name "venues", :schema "test-data", :description nil}))
+          (is (contains? (:tables actual-tables) {:name "checkins", :schema "test-data", :description nil}))
+          (is (contains? (:tables actual-tables) {:name "airport", :schema "airports", :description nil}))
+          (is (contains? (:tables actual-tables) {:name "bird", :schema "bird-flocks", :description nil})))
```
Comment (metamben) [inline modules/drivers/databricks/test/metabase/driver/databricks_test.clj:34]: Could use `are`.

File: modules/drivers/databricks/test/metabase/driver/databricks_test.clj [hunk 5]
```diff
@@ -21,7 +22,27 @@
...
+        (testing "information_schema is excluded"
+          (is (empty? (filter #(= "information_schema" (:name %)) (:tables actual-tables)))))))
```
Comment (metamben) [inline modules/drivers/databricks/test/metabase/driver/databricks_test.clj:36]: I find `(not-any? #(...) ...)` more direct. Why `:name` and not `:schema`?

File: modules/drivers/databricks/test/metabase/driver/databricks_test.clj [hunk 6]
```diff
@@ -21,7 +22,27 @@
...
+    (testing "`driver/describe-database` returns expected results for `exclusion` schema filters."
+      (let [actual-tables (driver/describe-database :databricks (update (mt/db) :details assoc
+                                                                        :schema-filters-patterns "test-data"
+                                                                        :schema-filters-type "exclusion"))]
+        (testing "tables from multiple schemas were found"
+          (is (not (contains? (:tables actual-tables) {:name "venues", :schema "test-data", :description nil})))
+          (is (not (contains? (:tables actual-tables) {:name "checkins", :schema "test-data", :description nil})))
```
Comment (metamben) [inline modules/drivers/databricks/test/metabase/driver/databricks_test.clj:43]: I would prefer a single assertion that `:schema "test-data"` doesn't occur.

**Review by metamben — COMMENTED** [review]
I think there is a typo there, otherwise LGTM.

**Review by metamben — APPROVED** [review]

File: modules/drivers/databricks/src/metabase/driver/databricks.clj (hunk 3 above)
Comment (lbrdnk) [reply modules/drivers/databricks/src/metabase/driver/databricks.clj:70]: Addressed in the follop-up PR.

File: modules/drivers/databricks/test/metabase/driver/databricks_test.clj (hunk 4 above)
Comment (lbrdnk) [reply modules/drivers/databricks/test/metabase/driver/databricks_test.clj:34]: Addressed in the follow-up PR.

File: modules/drivers/databricks/test/metabase/driver/databricks_test.clj (hunk 5 above)
Comment (lbrdnk) [reply modules/drivers/databricks/test/metabase/driver/databricks_test.clj]: Thanks! Addressed in this PR.

File: modules/drivers/databricks/test/metabase/driver/databricks_test.clj (hunk 6 above)
Comment (lbrdnk) [reply modules/drivers/databricks/test/metabase/driver/databricks_test.clj:43]: Addressed in the follow-up PR.

**Pull request merged**